File: baselib/src/main/java/com/wan/baselib/utils/ResourcesUtils.java
package com.wan.baselib.utils;

import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.drawable.Drawable;

import androidx.annotation.ColorInt;
import androidx.core.content.ContextCompat;

import com.wan.baselib.app.App;

/**
 * @author cy
 */
public class ResourcesUtils {
    /**
     * 获取strings.xml资源文件字符串
     *
     * @param id 资源文件id
     * @return 资源文件对应字符串
     */
    public static String getString(int id) {
        return App.context().getResources().getString(id);
    }

    /**
     * 获取strings.xml资源文件字符串数组
     *
     * @param id 资源文件id
     * @return 资源文件对应字符串数组
     */
    public static String[] getStringArray(int id) {
        return App.context().getResources().getStringArray(id);
    }

    /**
     * 获取drawable资源文件图片
     *
     * @param id 资源文件id
     * @return 资源文件对应图片
     */
    public static Drawable getDrawable(int id) {
        return ContextCompat.getDrawable(App.context(), id);
    }

    /**
     * 获取colors.xml资源文件颜色
     *
     * @param id 资源文件id
     * @return 资源文件对应颜色值
     */
    public static @ColorInt int getColor(Context context, int id) {
        return ContextCompat.getColor(context, id);
    }

    /**
     * 获取颜色的状态选择器
     *
     * @param id 资源文件id
     * @return 资源文件对应颜色状态
     */
    public static ColorStateList getColorStateList(int id) {
        return ContextCompat.getColorStateList(App.context(), id);
    }

    /**
     * 获取dimens资源文件中具体像素值
     *
     * @param id 资源文件id
     * @return 资源文件对应像素值
     */
    public static int getDimen(int id) {
        return App.context().getResources().getDimensionPixelSize(id);
    }

}

File: baselib/src/main/java/com/wan/baselib/utils/ResourcesUtils.kt
package com.wan.baselib.utils

import android.content.res.ColorStateList
import android.graphics.drawable.Drawable
import androidx.annotation.*
import splitties.init.appCtx
import splitties.resources.*

/**
 * @author cy
 */
object ResourcesUtils {

    /**
     * 获取strings.xml资源文件字符串
     *
     * @param id 资源文件id
     * @return 资源文件对应字符串
     */
    @JvmStatic
    fun getString(@StringRes id: Int): String {
        return appCtx.str(id)
    }

    /**
     * 获取strings.xml资源文件字符串数组
     *
     * @param id 资源文件id
     * @return 资源文件对应字符串数组
     */
    fun getStringArray(@ArrayRes id: Int): Array<String> {
        return appCtx.strArray(id)
    }

    /**
     * 获取drawable资源文件图片
     *
     * @param id 资源文件id
     * @return 资源文件对应图片
     */
    fun getDrawable(@DrawableRes id: Int): Drawable? {
        return appCtx.drawable(id)
    }

    /**
     * 获取colors.xml资源文件颜色
     *
     * @param id 资源文件id
     * @return 资源文件对应颜色值
     */
    @ColorInt
    fun getColor(@ColorRes id: Int): Int {
        return appCtx.color(id)
    }

    /**
     * 获取颜色的状态选择器
     *
     * @param id 资源文件id
     * @return 资源文件对应颜色状态
     */
    fun getColorStateList(@ColorRes id: Int): ColorStateList {
        return appCtx.colorSL(id)
    }

    /**
     * 获取dimens资源文件中具体像素值
     *
     * @param id 资源文件id
     * @return 资源文件对应像素值
     */
    fun getDimen(@DimenRes id: Int): Int {
        return appCtx.dimenPxSize(id)
    }
}